packages/affiliates/test/mocha/models.js

```diff
@@ -71,15 +72,46 @@ describe("History", function() {
   });
 });
 
-describe("Prices", function() {
+describe("Contract Prices", function() {
   let prices, seed;
   // Sample data from known contract.
   const token = "0xD16c79c8A39D44B2F3eB45D2019cd6A42B03E2A9";
   const startingTimestamp = moment("2020-09-23 23:00:00", "YYYY-MM-DD  HH:mm Z").valueOf();
   const endingTimestamp = moment("2020-10-05 23:00:00", "YYYY-MM-DD  HH:mm Z").valueOf();
   it("init", async function() {
-    seed = await Coingecko().chart(token, "usd", startingTimestamp, endingTimestamp);
-    prices = Prices(seed.prices);
+    seed = await Coingecko().getHistoricContractPrices(token, "usd", startingTimestamp, endingTimestamp);
+    prices = Prices(seed);
+    assert.ok(seed);
+    assert.ok(prices);
+  });
+  it("lookup", function() {
+    const time = moment()
+      .subtract(5, "days")
+      .valueOf();
+    const result = prices.lookup(time);
+    assert.ok(result[0] <= time);
+  });
+  it("closest", function() {
+    const time = moment()
+      .subtract(5, "days")
+      .valueOf();
+    const result = prices.closest(time);
+    assert.ok(result[0]);
+    assert.ok(result[1]);
+  });
+});
+
+describe("Synthetic prices", function() {
+  let prices, seed;
+  const empAddress = "0x3605Ec11BA7bD208501cbb24cd890bC58D2dbA56";
+  const startingTimestamp = moment("2020-09-23 23:00:00", "YYYY-MM-DD  HH:mm Z").valueOf();
+  const endingTimestamp = moment("2020-10-05 23:00:00", "YYYY-MM-DD  HH:mm Z").valueOf();
+  it("init", async function() {
+    this.timeout(100000);
+    const Web3 = require("web3");
+    const web3 = new Web3(new Web3.providers.HttpProvider(process.env.CUSTOM_NODE_URL));
```

Comment:
I'm not sure what the best way is to have this in the tests? The `SyntheticPrices` constructor requires a web3 instance connected to mainnet. Thoughts?

Reply:
yea this kind of sucks. Either you create a truffle test, which I think one of the previous prs adds test/truffle and then mock everything out to a local chain, or we can add it to e2e tests which dont get run in CI at the moment and just run locally. Imo its probably good enough to just run it locally to verify.  Not sure of what else we can do here without refactoring the underlying module.

Reply:
I've addresses this by leveraging the `getWeb3` function to generate a web3 instance. I also modified the `package.json` to run the `mocks tests` with a `--network mainnet_mnemonic` such that the `getWeb3` will connect to main net by default. This test file (`models.js`) pulls main net info for this test as well as `Coingecko().getHistoricContractPrices` to validate the price feed data so its fine that it's connecting to main net.